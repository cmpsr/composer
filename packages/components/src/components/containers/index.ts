export * from './Box';
export * from './Grid';
export * from './Section';
export * from './SectionChild';
