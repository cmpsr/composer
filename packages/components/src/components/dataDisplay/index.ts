export { ListItem } from '@chakra-ui/react';
export * from './Badge';
export * from './Divider';
export * from './NumberedList';
export * from './Symbol';
export * from './Tag';
export * from './UnorderedList';
