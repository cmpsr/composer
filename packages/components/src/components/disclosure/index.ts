export * from './Accordion';
export * from './Tabs';
