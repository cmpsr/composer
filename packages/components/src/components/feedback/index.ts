export * from './Alert';
export * from './InlineAlert';
export * from './ProgressBar';
export * from './ProgressCircular';
export * from './Skeleton';
export * from './Spinner';
export * from './Toast';
