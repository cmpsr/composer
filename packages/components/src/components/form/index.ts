export * from './Autocomplete';
export * from './AutocompleteMultiSelect';
export * from './Button';
export * from './ButtonIcon';
export * from './Checkbox';
export * from './DateInput';
export * from './IconButton';
export * from './Input';
export * from './MaskInput';
export * from './NumberInput';
export * from './PhoneInput';
export * from './PinInput';
export * from './Radio';
export * from './RangeSlider';
export * from './SegmentedButton';
export * from './Select';
export * from './Slider';
export * from './Switch';
export * from './Textarea';
