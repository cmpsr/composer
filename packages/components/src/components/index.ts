export * from './layouts';
export * from './primitives';
export * from './navigation';
export * from './notifications';
