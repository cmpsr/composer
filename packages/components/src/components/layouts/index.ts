export * from './Box';
export * from './Flex';
export * from './Grid';
export * from './Section';
