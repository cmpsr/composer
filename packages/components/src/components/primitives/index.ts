export * from './Accordion';
export * from './Alert';
export * from './Avatar';
export * from './Badge';
export * from './Breadcrumb';
export * from './Button';
export * from './Card';
export * from './CircularProgress';
export * from './Checkbox';
export * from './Divider';
export * from './Image';
export * from './Icons';
export * from './Input';
export * from './Link';
export * from './Progress';
export * from './Radio';
export * from './Select';
export * from './Slider';
export * from './Spinner';
export * from './Switch';
export * from './Tabs';
export * from './Tag';
export * from './Text';
export * from './TextPairing';
export * from './Textarea'
