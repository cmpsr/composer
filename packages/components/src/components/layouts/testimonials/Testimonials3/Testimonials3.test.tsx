import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Testimonials3, TESTIMONIAL3_PLAN_DEFAULT_TEST_ID } from '.';
import { IMAGE_DEFAULT_TEST_ID } from 'components/primitives';

describe('Testimonials3', () => {
  const asset = {
    contentType: 'image',
    title: 'Image Title',
    url:
      'https://images.ctfassets.net/o9153kt66j4s/6Ru9unmOhMXrFwKtLV2SRU/7c902fd189741be40f5fef77340cbf1d/placeholder.svg',
  };
  const testimonial =
    'Lorem ipsum dolor sit amet consectetur adipisicing elit. Repudiandae aut, omnis, maxime rem, cum similique repellendus hic impedit totam magnam deserunt voluptatum quia! Blanditiis repellendus sit debitis doloribus, totam minima?';
  const author = 'Testing author,';
  const profession = 'Testing profession.';

  it('should match snapshot', () => {
    const testimonial3 = renderer.create(
      <Testimonials3
        asset={asset}
        testimonial={testimonial}
        author={author}
        profession={profession}
      />,
    );

    const tree = testimonial3.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render component', () => {
    const { getByTestId } = render(
      <Testimonials3
        asset={asset}
        testimonial={testimonial}
        author={author}
        profession={profession}
      />,
    );
    const testimonial3 = getByTestId(TESTIMONIAL3_PLAN_DEFAULT_TEST_ID);
    expect(testimonial3).toBeInTheDocument();
  });
  it('should render className', () => {
    const { getByTestId } = render(
      <Testimonials3
        asset={asset}
        testimonial={testimonial}
        author={author}
        profession={profession}
        className="foo"
      />,
    );

    const testimonial3 = getByTestId(TESTIMONIAL3_PLAN_DEFAULT_TEST_ID);
    expect(testimonial3.classList).toContain('foo');
  });
  it('should render rounded asset', () => {
    const { getByTestId } = render(
      <Testimonials3
        asset={asset}
        testimonial={testimonial}
        author={author}
        profession={profession}
      />,
    );

    const assetComponent = getByTestId(IMAGE_DEFAULT_TEST_ID);
    expect(assetComponent).toBeInTheDocument();
    expect(assetComponent.classList).toContain('rounded-full');
  });
  it('should render testimonial', () => {
    const { getByText } = render(
      <Testimonials3
        asset={asset}
        testimonial={testimonial}
        author={author}
        profession={profession}
      />,
    );
    const testimonial3 = getByText(testimonial);
    expect(testimonial3).toBeInTheDocument();
  });
  it('should render author', () => {
    const { getByText } = render(
      <Testimonials3
        asset={asset}
        testimonial={testimonial}
        author={author}
        profession={profession}
      />,
    );
    const testimonial3 = getByText(author);
    expect(testimonial3).toBeInTheDocument();
  });
  it('should render profession', () => {
    const { getByText } = render(
      <Testimonials3
        asset={asset}
        testimonial={testimonial}
        author={author}
        profession={profession}
      />,
    );
    const testimonial3 = getByText(profession);
    expect(testimonial3).toBeInTheDocument();
  });
});
