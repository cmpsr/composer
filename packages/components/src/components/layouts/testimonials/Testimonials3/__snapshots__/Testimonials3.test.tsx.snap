// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testimonials3 should match snapshot 1`] = `
<div
  className="flex flex-col items-center"
  data-testid="testimonial3"
>
  <picture
    className="select-none mb-4"
  >
    <source
      className="select-none w-32 h-32 rounded-full object-cover"
      data-testid="image"
      srcSet="https://images.ctfassets.net/o9153kt66j4s/6Ru9unmOhMXrFwKtLV2SRU/7c902fd189741be40f5fef77340cbf1d/placeholder.svg?fm=webp"
      type="image/webp"
    />
    <img
      alt="Image Title"
      className="select-none w-32 h-32 rounded-full object-cover"
      data-testid="imageContent"
      src="https://images.ctfassets.net/o9153kt66j4s/6Ru9unmOhMXrFwKtLV2SRU/7c902fd189741be40f5fef77340cbf1d/placeholder.svg"
    />
  </picture>
  <h6
    className="typo-headline-6"
    data-testid="typography"
  >
    Testing author,
  </h6>
  <p
    className="typo-body text-gray-500 mb-6"
    data-testid="typography"
  >
    Testing profession.
  </p>
  <p
    className="typo-body text-center md:px-5"
    data-testid="typography"
  >
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Repudiandae aut, omnis, maxime rem, cum similique repellendus hic impedit totam magnam deserunt voluptatum quia! Blanditiis repellendus sit debitis doloribus, totam minima?
  </p>
</div>
`;
